AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSHTTPURLResponse.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// `AWSHTTPURLResponse` contains the response body and metadata associated with the response to an HTTP request.
+///
+/// When using `AWSAPIPlugin`, you can optionally type cast `HTTPURLResponse` to an instances of `AWSHTTPURLResponse`.
+/// The response body can be accessed from the `body: Data?` property. For example, when the `APIError` is an
+/// `.httpStatusError(StatusCode, HTTPURLResponse)`, then access the response body by type casting the response to
+/// an `AWSHTTPURLResponse` and retrieve the `body` field.
+/// ```swift
+/// if case let .httpStatusError(statusCode, response) = error, let awsResponse = response as? AWSHTTPURLResponse {
+///     if let responseBody = awsResponse.body {
+///         print("Response contains a response body")
+///     }
+/// }
+/// ```
+/// **Note**: The class inheritance to `HTTPURLResponse` is to provide above mechanism, and actual
+/// implementation acts as a facade that stores an instance of `HTTPURLResponse` that delegates overidden methods to
+/// this stored property.
+public class AWSHTTPURLResponse: HTTPURLResponse {
+
+    /// The response body of the response
+    public let body: Data?
+
+    private let response: HTTPURLResponse
+
+    init?(response: HTTPURLResponse, body: Data?) {
+        self.body = body
+        self.response = response
+
+        // Call the super class initializer with dummy values to satisfy the requirement of the inheritance.
+        super.init(url: URL(string: "dummyURL")!,
```

Comment:
Why can't we call this with `response.url` and only default to the dummy if that isn't present?

Reply:
we can, but given the implementation is to proxy the calls to the stored property, I did not want to introduce logic that could confuse the reader such as "why are we storing passing the `response.url` to the super.init` but still store the httpURLResponse?